components/VM_Arm/CMakeLists.txt

```diff
@@ -127,6 +127,20 @@ config_option(
     OFF
 )
 
+config_option(
+    VmCustomDtbBaseFile
+    VM_CUSTOM_DTB_BASE_FILE
+    "Use a custom device tree base when generating a guest's device tree. This can be useful when \
+    the device tree needed by the guest is significantly different than the device tree provided by \
+    the kernel. For example, the TX2 bootloader modifies the device tree, so the tx2.dts in the \
```

Comment:
Maybe clarify this comment a bit more when it comes to the phrase "device tree provided by the kernel" and say "provided by the platform" . As far as I know, we compile the kernel against a given devices tree file. When booting in the platform, the device tree passed is the one from the platform bootloader, which can be different already to the one the kernel was compiled with. So the actual problem might start by a wrong device tree used during kernel compilation or missing sanity check on the ElfLoader or kernel, that things still match. What is the actual issue you are running into, is the memory configuration different because the boot loader carves out memory dynamically? In the worst case we might have much bigger problem on the TX2  platform then...?

Reply:
I'll make that change.

There's also the ZCU102, which in 2021.1 (?) updated the bus from `amba` to `axi`, so if you generate a device tree from the `zynqmp.dts` in the kernel while using a 2021.1 kernel, things won't work. But with this change you could keep the kernel unmodified and pass a new device tree to the VM to generate against.

As for the TX2, I was unaware (and maybe things have been updated since I worked with the platform) that the `tx2.dts` in the kernel is _not_ the device tree used as a base for the generation.